splink/diagnostics.py

```diff
@@ -0,0 +1,122 @@
+from pyspark.sql.dataframe import DataFrame
+from pyspark.sql.session import SparkSession
+from pyspark.context import SparkContext, SparkConf
+from pyspark.sql import SQLContext
+
+from typing import Callable
+
+from pyspark.ml.regression import LinearRegression
+from pyspark.ml.linalg import DenseVector
+from pyspark.ml.linalg import Vectors
+from pyspark.ml.evaluation import RegressionEvaluator
+
+from pyspark.sql.types import StructType, StructField, StringType, DoubleType, Row
+import pyspark.sql.functions as f
+from pyspark.sql.functions import when
+from .check_types import check_types
+import warnings
+
+
+@check_types
+def vif_gammas(df_gammas: DataFrame, spark: SparkSession, sampleratio: float = 1.0):
+
+    """splink diagnostic of multicollinearity in gamma values
+    
+    We want to check if  the gammas  of the input variables of the models we are using suffer from multicollinearity. 
+    Since the columns information is transformed into  gammas , and it is the information from the gammas 
+    which the model 'sees', therefore it is corelation on the gammas which is potentially problematic.
+    
+    
+    
+        Args:
+        
+            df_gammas (DataFrame): A dataframe of record comparisons containing gamma comaprisno values, 
+            e.g. as produced by the splink.gammas.add_gammas function
+            
+            spark (SparkSession): SparkSession object
+            
+            sampleratio (float,optional) : Fraction of rows to sample, range [0.0, 1.0]. 
+            If 1.0 no sampling takes place
+            
+            
+        
+        Returns:
+            
+            (DataFrame) : Spark dataframe of the gamma variables Variance Inflation Factors (VIFs)
+            
+        
+        """
+
+    collist = []
+    viflist = []
+
+    sc = spark.sparkContext
+    sqlContext = SQLContext(sc)
```

Comment:
I think sqlContext here is the same thing as `spark` e.g. later i think you can just do `spark.createDataFrame`. iirc

Reply:
corrected this 👍

Reply:
ok. now it uses spark (sparkSession)